clarity/src/vm/tests/principals.rs

```diff
@@ -284,3 +289,815 @@ fn test_simple_is_standard_undefined_cases() {
+        Value::Response(ResponseData {
+            committed: true,
+            data: Box::new(Value::Principal(PrincipalData::Contract(
+                QualifiedContractIdentifier::new(
+                    StandardPrincipalData(26, transfer_buffer),
+                    "hello-world".try_into().unwrap()
+                )
+            )))
+        }),
+        execute_with_parameters(
+            input,
+            ClarityVersion::Clarity2,
+            StacksEpochId::Epoch21,
+            false
+        )
+        .unwrap()
+        .unwrap()
+    );
+
+    // Testnet multi-sig contract, run on testnet.
+    let input =
+        r#"(principal-construct 0x15 0xfa6bf38ed557fe417333710d6033e9419391a320 "hello-world")"#;
+    assert_eq!(
+        Value::Response(ResponseData {
+            committed: true,
+            data: Box::new(Value::Principal(PrincipalData::Contract(
+                QualifiedContractIdentifier::new(
+                    StandardPrincipalData(21, transfer_buffer),
+                    "hello-world".try_into().unwrap()
+                )
+            )))
+        }),
+        execute_with_parameters(
+            input,
+            ClarityVersion::Clarity2,
+            StacksEpochId::Epoch21,
+            false
+        )
+        .unwrap()
+        .unwrap()
+    );
+}
+
+/// Creates a `Principal`-type `Value` from string-based byte representations.
+fn create_principal_from_strings(
+    version_string: &str,
+    principal_string: &str,
+    name: Option<&str>,
+) -> Value {
+    let mut version_array = [0u8; 1];
+    version_array.copy_from_slice(&hex_bytes(version_string).expect("hex_arrays failed"));
+    let mut principal_array = [0u8; 20];
+    principal_array.copy_from_slice(&hex_bytes(principal_string).expect("hex_bytes failed"));
+
+    if let Some(name) = name {
+        // contract principal requested
+        Value::Principal(PrincipalData::Contract(QualifiedContractIdentifier::new(
+            StandardPrincipalData(version_array[0], principal_array),
+            name.try_into().unwrap(),
+        )))
+    } else {
+        // standard principal requested
+        Value::Principal(PrincipalData::Standard(StandardPrincipalData(
+            version_array[0],
+            principal_array,
+        )))
+    }
+}
+
+#[test]
+// Test cases where the version byte is of the right type `(buff 1)`, but where the byte doesn't
+// match a recognized network. This is meant for compatibility with "future" network bytes, so
+// is still valid.
+fn test_principal_construct_version_byte_future() {
+    // The version byte 0x1f is unrecognized today, but is valid for the future.
+    let input = r#"(principal-construct 0x1f 0x0102030405060708091011121314151617181920)"#;
+    assert_eq!(
+        Value::Response(ResponseData {
+            committed: false,
+            data: Box::new(Value::Tuple(
+                TupleData::from_data(vec![
+                    ("error_int".into(), Value::UInt(0 as u128)),
```

Comment:
Once you create constants for the error codes you can use them in these tests too

Reply:
Fixed